### PR TITLE
(chore) - remove focusElement

### PR DESCRIPTION
**Pull request by JoviDeCroock**
Adding items in react: https://codesandbox.io/s/748j0p3jpj
Swapping in react: https://codesandbox.io/s/olq48z68r5

So the first we should support, which also feels obviosu to me personally but the second not so much.

The odd thing is that it only  fails when you insert a second thing above

So add one before --> works
Add one after --> works
Add one after previous --> works
Add one after the first step --> fails

Size of core stays the same
Seems to give a significant performance increase

### CODE REVIEW

**Comment by JoviDeCroock** [comment]
Don't take the Passing as it succeeds I have adjusted this in tests:
https://github.com/developit/preact/pull/1548/files#diff-4860f971b6d2f661624c4d35fbba8cc3L106

Now it effectively tests with keys.

Without keys it loses focus when the last insertion happens so insert 0 before 1.
Then we treat input like a new element, logging output:

```
LOG: 'parent before', <div><span>0</span><input type="text"><span>1</span><span>3</span></div>, <input type="text">, <span>3</span>
LOG: 'parent append', <span></span>, 3
LOG: 'parent append', <div><span>0</span><span>1</span><input type="text"><span>2</span></div>, <span>3</span>
```

'parent before' is first the existing parentDom and then adding a newDom before given oldDom
'parent append' adds the second argument to the first

File: src/diff/children.js [hunk 1]
```diff
@@ -42,6 +43,7 @@ export function diffChildren(parentDom, newParentVNode, oldParentVNode, context,
 			for (i = 0; i < excessDomChildren.length; i++) {
 				if (excessDomChildren[i]!=null) {
 					oldDom = excessDomChildren[i];
+					oldChild = excessDomChildren[i];
```
Comment (JoviDeCroock) [inline src/diff/children.js]: This can be removed.

File: test/browser/focus.test.js [hunk 2]
```diff
@@ -54,7 +54,7 @@ describe('focus', () => {
 		teardown(scratch);
 	});
 
-	it('should maintain focus when swapping elements', () => {
+	it.skip('should maintain focus when swapping elements', () => {
```
Comment (ForsakenHarmony) [inline test/browser/focus.test.js]: should this be skipped?

Comment (JoviDeCroock) [reply test/browser/focus.test.js]: Yes, Swapping in react: https://codesandbox.io/s/olq48z68r5 this is not supported in most frameworks and can lead to unwanted behavior imo

Comment (marvinhagemeister) [reply test/browser/focus.test.js]: @JoviDeCroock The linked sandbox just appends new `h1` elements for me

Comment (JoviDeCroock) [reply test/browser/focus.test.js]: It swaps the input from before to after the h1 and that loses focus on the input or am I misunderstanding it? @marvinhagemeister

Comment (ForsakenHarmony) [reply test/browser/focus.test.js]: wait am I missing something there? what does it swap?

Comment (JoviDeCroock) [reply test/browser/focus.test.js]: it swaps the input to bbefore the H1 same behavior as the test that is skipped?

Comment (ForsakenHarmony) [reply test/browser/focus.test.js]: The h1 stays?

Comment (marvinhagemeister) [reply test/browser/focus.test.js]: Maybe you forgot to save the sandbox. Re-created it where the elements are swapped all the time: https://codesandbox.io/s/0yo8kxqjzn . Can confirm that `react` loses focus there, too 👍

**Comment by ForsakenHarmony** [comment]
🎉

**Pull request merged**